catroid/src/androidTest/java/org/catrobat/catroid/test/embroidery/DSTFileGeneratorTest.java

```diff
@@ -0,0 +1,203 @@
+public class DSTFileGeneratorTest {
+	final String projectName = "testProject";
+	DSTFileGenerator fileGenerator;
+	File dstFile;
+	final int[] conversion = {-81, 81, -27, 27, -9, 9, -3, 3, -1, 1};
+
+	@Before
+	public void setUp() {
+		Project project = new Project(InstrumentationRegistry.getTargetContext(), projectName);
+		ProjectManager.getInstance().setCurrentProject(project);
+		fileGenerator = new DSTFileGenerator();
+		dstFile = new File(Constants.CACHE_DIR + "/" + projectName + ".dst");
+	}
+
+	private int getValueForByte(int element) {
+		int mask = 0x200;
+		int value = 0;
+		for (int i = 0; i < conversion.length; i++) {
+			if ((element & (mask >>> i)) != 0) {
+				value += conversion[i];
+			}
+		}
+		return value;
+	}
+
+	@Test
+	public void testConversionTable() {
+		for (int element = 0; element < DSTFileGenerator.CONVERSIONTABLE.length; element++) {
+			int value = getValueForByte(DSTFileGenerator.CONVERSIONTABLE[element]);
+			if (element > 121) {
+				assertEquals((element - 121) * (-1), value);
+			} else {
+				assertEquals(element, value);
+			}
+		}
+	}
+
+	@Test
+	public void testWriteToSampleDSTFile() throws IOException, URISyntaxException {
+		EmbroideryList stitchpoints = new EmbroideryList();
+		stitchpoints.add(new PointF(-10, 0));
+		stitchpoints.add(new PointF(10, 0));
+		stitchpoints.add(new PointF(10, 10));
+		stitchpoints.add(new PointF(0, 15));
+		stitchpoints.add(new PointF(-10, 10));
+		stitchpoints.add(new PointF(-10, 0));
+		stitchpoints.add(new PointF(10, 10));
+		stitchpoints.add(new PointF(-10, 10));
+		stitchpoints.add(new PointF(10, 0));
+		if (dstFile.exists()) {
+			dstFile.delete();
+		}
+		dstFile.createNewFile();
+		fileGenerator.writeToDSTFile(dstFile, stitchpoints);
+
+		InputStream inputStream = InstrumentationRegistry.getContext().getResources().openRawResource(org.catrobat
+				.catroid.test.R.raw.sample_dst_file);
+		File compareFile = StorageOperations.copyStreamToDir(inputStream, Constants.CACHE_DIR, "sample_dst_file.dst");
+
+		assertEquals(compareFile.length(), dstFile.length());
+
+		byte[] compareFileBytes = FileUtils.readFile(compareFile);
+		byte[] dstFileBytes = FileUtils.readFile(dstFile);
+
+		assertArrayEquals(compareFileBytes, dstFileBytes);
+	}
+
+	@Test
+	public void testPrepareStitchPoints() {
+		final int expectedAmountJumpPoints = 2;
+		final DSTFileGenerator.StitchPoint[] expectedStitchpoints = new DSTFileGenerator.StitchPoint[]{
+				new DSTFileGenerator.StitchPoint(new PointF(0, 0)),
+				new DSTFileGenerator.StitchPoint(new PointF(0, 0), true),
+				new DSTFileGenerator.StitchPoint(new PointF(100, 100), true),
+				new DSTFileGenerator.StitchPoint(new PointF(200, 200))};
+
+		EmbroideryList points = new EmbroideryList();
+		points.add(new PointF(0, 0));
+		points.add(new PointF(100, 100));
+		ArrayList<DSTFileGenerator.StitchPoint> stitchPoints = fileGenerator.prepareStitchPoints(points);
+
+		assertEquals(points.size() + expectedAmountJumpPoints, stitchPoints.size());
```

Comment:
please remove this, if this isnt true, the next assertion will fail anyways and also give you a pointer on why the length is wrong.